lib/winapi/libloaderapi.c

```diff
@@ -30,9 +31,46 @@ BOOL FreeLibrary (HMODULE hLibModule)
     return TRUE;
 }
 
+static PIMAGE_EXPORT_DIRECTORY find_edataxb (void)
+{
+    DWORD num_sections = CURRENT_XBE_HEADER->NumberOfSections;
+    PXBE_SECTION_HEADER section_header_addr = CURRENT_XBE_HEADER->PointerToSectionTable;
+
+    for (DWORD i = 0; i < num_sections; i++) {
+        if (memcmp(section_header_addr[i].SectionName, ".edataxb", 8) == 0) {
+            return (PIMAGE_EXPORT_DIRECTORY)section_header_addr[i].VirtualAddress;
+        }
+    }
+
+    return NULL;
+}
+
 FARPROC GetProcAddress (HMODULE hModule, LPCSTR lpProcName)
 {
-    // FIXME: If hModule is NULL, the symbol is looked up in the current module
+    if (hModule == NULL) {
+        // If hModule is NULL, the symbol is looked up in the main module
```

Comment:
This comment is odd. It does not proper describe why this is happening but how(which I can read myself). Maybe change it to something like: 
"When the dll handle is not given, retrieve the address of an exported function in the calling module"

Reply:
`When the dll handle is not given` is logically equivalent to `If hModule is NULL`, but I went ahead and changed it anyway. The second part of the comment is the more important one.

`retrieve the address of an exported function in the calling module` is not correct, the module in which the symbol is looked up is the main module (the xbe, in our case), not the calling module (which could be a dll in the future). It's a method used on Windows to look up exe symbols from a dll that was loaded by the exe.